platforms/nuttx/src/px4/common/usr_hrt.cpp

```diff
@@ -121,6 +107,43 @@ event_thread(int argc, char *argv[])
 	return 0;
 }
 
+static void start_worker(void)
+{
+	uintptr_t started;
+
+	// Don't use g_usr_hrt_task as the "lock" indicator, as this can result in
+	// two threads trying to create the worker
+	sched_lock();
+	started = g_worker_started;
+	g_worker_started = 1;
+	sched_unlock();
+
+	if (!started) {
+		g_usr_hrt_task = px4_task_spawn_cmd("usr_hrt", SCHED_DEFAULT, SCHED_PRIORITY_MAX,
```

Comment:
Can this fail? maybe if it fails, it would need to be reflected again in g_worker_started? Don't know.

Reply:
Of course it can fail. But I think the usr_hrt module is in deep doodoo in that case regardless.

Better yet, how should the failure be handled ? Attempt again and again ?

Reply:
Yes, I guess no point to try to handle it